scripts/sd_tag_batch.py

```diff
@@ -40,17 +83,24 @@ def run(self, p, in_front, prompt_weight, use_deepbooru, use_weight):
             interrogator = deepbooru.model.tag(p.init_images[0])
         else:
             interrogator = shared.interrogator.interrogate(p.init_images[0])
+        
+        # Remove duplicate prompt content from interrogator prompt
+        if no_duplicates:
+            interrogator = self.filter_words(interrogator, raw_prompt)
+        # Remove negative prompt content from interrogator prompt
+        if use_negatives:
+            interrogator = self.filter_words(interrogator, p.negative_prompt)
 
         if use_weight:
             if p.prompt == "":
-                p.prompt = Script.interrogator
+                p.prompt = interrogator
```

Comment:
Since there was no Script.interrogator, in the current commit, I was getting a Script.interrogator does not exist error, I noticed that interrogator was the variable being used to hold the interrogator string, which was previously handled by Script.interrogator. So I updated every instance of Script.interrogator variable, which does not exist, to the interrogator variable.